### PR TITLE
Remove support for Python 3.8 (& fix failing MacOs 3.9 tests)

### PR DESCRIPTION
**Pull request by shaunagm**
Not sure what to do about the test failures of 3.9 and higher, but we can treat them separately.

For the record, there appear to be two distinct issues, a slightly older one where the combo of Mac, Python 3.9, and limited dependencies leads to [a psycopg2 install error](https://github.com/move-coop/parsons/actions/runs/11957548537/job/33334963382?pr=1192#step:5:24) and then one that's an issue for all limited dependency installs related to [this pydantic release](https://github.com/pydantic/pydantic/issues/10910) - the latter issue will hopefully have an easy fix soon but I'm not sure about the former.

### CODE REVIEW

**Comment by shaunagm** [comment]
This should close #1138 and #1066 

Would love review on [this workaround](https://github.com/move-coop/parsons/pull/1192/files#diff-3e744f163053a3e58aeec8dede7e3711d8a87b2249a31fd94a884200b0aec1c7L34) - the tests are passing now but I'm not an installation expert and want to double check I'm not missing anything

File: parsons/etl/etl.py [hunk 1]
```diff
@@ -659,20 +658,12 @@ def unpack_nested_columns_as_rows(self, column, key="id", expand_original=False)
...
-                        lambda row: hashlib.md5(  # nosec B324
-                            str.encode("".join([str(x) for x in row]))
-                        ).hexdigest(),
-                    )
+                orig.add_column(
+                    "uid",
+                    lambda row: hashlib.md5(
+                        str.encode("".join([str(x) for x in row])), usedforsecurity=False
+                    ).hexdigest(),
+                )
```
Comment (austinweisgrau) [inline parsons/etl/etl.py:666]: IDK what this section of code is, but these changes seem out of scope of this PR?

File: .github/workflows/python-checks.yml [hunk 2]
```diff
@@ -160,6 +164,10 @@
                   python-version: ${{ matrix.python-version }}
                   cache: pip
 
+            - name: Patch install error when using Python 3.9, limited dependencies, and MacOS
+              if: ${{ matrix.limited-dependencies }} == True and ${{ matrix.os }} == "macos-latest" and ${{ matrix.python-version }} == "3.9"
+              run: pip install --system psycopg2-binary==2.9.9
+
```
[the remark this answers is not in the document]
Comment (austinweisgrau) [reply .github/workflows/python-checks.yml:170]: this seems solid and makes sense to me

File: parsons/etl/etl.py (hunk 1 above)
Comment (austinweisgrau) [reply parsons/etl/etl.py:666]: ohh I see, it's checking for the python version being 3.9 or higher

**Review by austinweisgrau — APPROVED** [review]
i like it
